futures-sink/src/lib.rs

```diff
@@ -114,101 +114,70 @@ pub trait Sink {
     /// The type of value produced by the sink when an error occurs.
     type SinkError;
 
-    /// Check if the sink is ready to start sending a value.
+    /// Attempts to prepare the `Sink` to receive a value.
+    ///
+    /// This method must be called and return `Ok(Async::Ready(()))` prior to
+    /// each call to `start_send`.
+    ///
+    /// This method returns `Async::Ready` once the underlying sink is ready to
+    /// receive data. If this method returns `Async::Pending`, the current task
+    /// is registered to be notified (via `cx.waker()`) when `poll_ready`
+    /// should be called again.
+    ///
+    /// In most cases, if the sink encounters an error, the sink will
+    /// permanently be unable to receive items.
     fn poll_ready(&mut self, cx: &mut task::Context) -> Poll<(), Self::SinkError>;
 
     /// Begin the process of sending a value to the sink.
+    /// Each call to this function must be proceeded by a successful call to
+    /// `poll_ready` which returned `Ok(Async::Ready(()))`.
     ///
     /// As the name suggests, this method only *begins* the process of sending
     /// the item. If the sink employs buffering, the item isn't fully processed
     /// until the buffer is fully flushed. Since sinks are designed to work with
     /// asynchronous I/O, the process of actually writing out the data to an
-    /// underlying object takes place asynchronously. **You *must* use
-    /// `flush` in order to drive completion of a send**. In particular,
-    /// `start_send` does not begin the flushing process
-    ///
-    /// # Return value
+    /// underlying object takes place asynchronously.**You *must* use
+    /// `poll_flush` or `poll_close` in order to guarantee completion of a
+    /// send**.
     ///
     /// This method returns `AsyncSink::Ready` if the sink was able to start
     /// sending `item`. In that case, you *must* ensure that you call
```

Comment:
This is feeling a bit repetitive, since you just said (and highlighted!) the flushing requirement. Also this only talks about flush rather than also including close as above.

Reply:
Done